pallets/schemas/src/lib.rs

```diff
@@ -790,6 +1238,100 @@ pub mod pallet {
+					payload_location == PayloadLocation::Paginated,
+				Error::<T>::InvalidSetting
+			);
+			let parsed_name = Self::parse_and_verify_new_name(&intent_name_payload)?;
+			let intent_id = Self::add_intent(payload_location, settings, &parsed_name)?;
+			Ok((intent_id, parsed_name))
+		}
+
+		/// Adds a given Intent to storage. If the pallet's maximum Intent limit has been
+		/// fulfilled by the time this extrinsic is called, an IntentCountOverflow error
+		/// will be thrown.
+		///
+		/// # Errors
+		/// * [`Error::IntentGroupCountOverflow`] - The schema count has exceeded its bounds
+		/// * [`Error::NameAlreadyExists`] - The name already exists
+		/// * [`Error::InvalidSchemaNameEncoding`] - The name has an invalid encoding
+		/// * [`Error::InvalidSchemaNameCharacters`] - The name contains invalid characters
+		/// * [`Error::InvalidSchemaNameStructure`] - The name has an invalid structure (i.e., not `protocol.descriptor`)
+		/// * [`Error::InvalidSchemaNameLength`] - The name exceed the allowed overall name length
+		/// * [`Error::InvalidSchemaNamespaceLength`] - The protocol portion of the name exceeds the max allowed length
+		/// * [`Error::InvalidSchemaDescriptorLength`] - The descriptor portion of the name exceeds the max allowed length
+		/// * [`Error::InvalidIntentId`] - One of the [`IntentId`]s specified does not exist
+		pub fn create_intent_group_for(
+			intent_group_name: SchemaNamePayload,
+			intent_ids: BoundedVec<IntentId, T::MaxIntentsPerIntentGroup>,
+		) -> Result<(IntentGroupId, SchemaName), DispatchError> {
+			let intent_group_name = Self::parse_and_verify_new_name(&intent_group_name)?;
+			Self::validate_intent_ids(&intent_ids)?;
+			let intent_id = Self::add_intent_group(&intent_group_name, intent_ids)?;
+			Ok((intent_id, intent_group_name))
+		}
+
+		/// Update the list of [`IntentId`]s associated with a given [`IntentGroupId`]
+		///
+		/// # Errors
+		/// * [`Error::InvalidIntentGroupId`] - The specified [`IntentGroupId`] doesn't exist
+		/// * [`Error::InvalidIntentId`] - One of the [`IntentId`]s specified doesn't exist
+		pub fn update_intent_group_for(
+			intent_group_id: IntentGroupId,
+			intent_ids: BoundedVec<IntentId, T::MaxIntentsPerIntentGroup>,
+		) -> Result<(), DispatchError> {
+			ensure!(
+				IntentGroups::<T>::contains_key(intent_group_id),
+				Error::<T>::InvalidIntentGroupId
+			);
+			Self::validate_intent_ids(&intent_ids)?;
+			Self::update_intent_group_storage(intent_group_id, intent_ids)?;
+			Ok(())
+		}
+
+		/// Validate that all items in a list of [`IntentId`]s exist
+		///
+		/// # Errors
+		/// * [`Error::InvalidIntentId`] - At least one of the specified [`IntentId`]s does not exist
+		///
+		pub fn validate_intent_ids(
+			intent_ids: &BoundedVec<IntentId, T::MaxIntentsPerIntentGroup>,
+		) -> Result<(), DispatchError> {
+			intent_ids.iter().try_for_each::<_, _>(|intent_id| {
+				ensure!(IntentInfos::<T>::contains_key(intent_id), Error::<T>::InvalidIntentId);
```

Comment:
Can we optimize this to only check against the `CurrentIntentIdentifierMaximum`? It's not like we are removing anything from IntentInfos?

Reply:
Good idea.